crates/sui-light-client/src/proof.rs

```diff
@@ -0,0 +1,230 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::anyhow;
+
+use sui_types::{
+    base_types::ObjectRef,
+    committee::Committee,
+    effects::{TransactionEffects, TransactionEffectsAPI, TransactionEvents},
+    event::{Event, EventID},
+    messages_checkpoint::{CertifiedCheckpointSummary, CheckpointContents, EndOfEpochData},
+    object::Object,
+    transaction::Transaction,
+};
+
+/// Define aspect of Sui state that need to be certified in a proof
+#[derive(Default)]
+pub struct ProofTarget {
+    /// Objects that need to be certified
+    pub objects: Vec<(ObjectRef, Object)>,
+
+    /// Events that need to be certified
+    pub events: Vec<(EventID, Event)>,
+
+    /// The next committee being certified
+    pub committee: Option<Committee>,
+}
+
+impl ProofTarget {
+    /// Create a new empty proof target. An empty proof target still ensures that the
+    /// checkpoint summary is correct.
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    /// Add an object to be certified by object reference and content. A verified proof will
+    /// ensure that both the reference and content are correct. Note that some content is
+    /// meta-data such as the transaction that created this object.
+    pub fn add_object(mut self, object_ref: ObjectRef, object: Object) -> Self {
+        self.objects.push((object_ref, object));
+        self
+    }
+
+    /// Add an event to be certified by event ID and content. A verified proof will ensure that
+    /// both the ID and content are correct.
+    pub fn add_event(mut self, event_id: EventID, event: Event) -> Self {
+        self.events.push((event_id, event));
+        self
+    }
+
+    /// Add the next committee to be certified. A verified proof will ensure that the next
+    /// committee is correct.
+    pub fn set_committee(mut self, committee: Committee) -> Self {
+        self.committee = Some(committee);
+        self
+    }
+}
+
+/// Part of a Proof that provides evidence relating to a specific transaction to
+/// certify objects and events.
+pub struct TransactionProof {
+    /// Checkpoint contents including this transaction.
+    pub checkpoint_contents: CheckpointContents,
+
+    /// The transaction being certified.
+    pub transaction: Transaction,
+
+    /// The effects of the transaction being certified.
+    pub effects: TransactionEffects,
+
+    /// The events of the transaction being certified.
+    pub events: Option<TransactionEvents>,
+}
+
+/// A proof for specific targets. It certifies a checkpoint summary and optionally includes
+/// transaction evidence to certify objects and events.
+pub struct Proof {
+    /// Targets of the proof are objects, events or a committee that need to be certified
```

Comment:
```suggestion
    /// Targets of the proof are a committee, objects, or events that need to be certified.
```
Suggest order list this way so "need" doesn't appear incorrect.